rs/cli/src/discourse_client.rs

```diff
@@ -149,32 +179,54 @@ impl DiscourseClientImp {
             update_id: None,
         }))
     }
+
+    fn request_from_user_post(&self, err: anyhow::Error, body: String, topic_url: String) -> anyhow::Result<Option<DiscourseResponse>> {
+        warn!("Received error: {:?}", err);
+        warn!("Please create a post in topic {} with the following content", topic_url);
+        println!("{}", body);
+        let forum_post_link = dialoguer::Input::<String>::new()
+            .with_prompt("Forum post link")
+            .allow_empty(true)
+            .interact()?;
+        Ok(Some(DiscourseResponse {
+            url: forum_post_link,
+            update_id: None,
+        }))
+    }
 }
 
 const GOVERNANCE_TOPIC: &str = "Governance";
 const SUBNET_MANAGEMENT_TAG: &str = "Subnet-management";
 
+#[derive(Deserialize)]
+struct SubnetTopicInfo {
+    slug: String,
+    topic_id: u64,
+}
+
+const SUBNET_TOPICS_AND_SLUGS: &str = include_str!("assets/subnet_topic_map.json");
+fn get_subnet_topics_map() -> BTreeMap<PrincipalId, SubnetTopicInfo> {
+    serde_json::from_str(SUBNET_TOPICS_AND_SLUGS).unwrap()
+}
+
 impl DiscourseClient for DiscourseClientImp {
     fn create_replace_nodes_forum_post(&self, subnet_id: PrincipalId, body: String) -> BoxFuture<'_, anyhow::Result<Option<DiscourseResponse>>> {
-        let subnet_id = subnet_id.to_string();
-        // All principals have a `-` in the string from
-        let (first_part, _rest) = subnet_id.split_once("-").unwrap();
-        let post = DiscourseTopic {
-            title: format!("Replacing nodes in subnet {}", first_part),
-            content: body,
-            tags: vec![SUBNET_MANAGEMENT_TAG.to_string()],
-            category: GOVERNANCE_TOPIC.to_string(),
-        };
-        let post_clone = post.clone();
-
-        let try_call = async move {
+        Box::pin(async move {
             if self.offline || self.skip_forum_post_creation {
                 return Ok(None);
             }
-            let topic = self.create_topic(post).await?;
-            Ok(Some(topic))
-        };
-        Box::pin(async move { try_call.await.or_else(|e| self.request_from_user(e, post_clone)) })
+
+            let subnet_topic_map = get_subnet_topics_map();
+            let topic_info = subnet_topic_map.get(&subnet_id).ok_or(anyhow::anyhow!(
```

Comment:
Would be good to let the user override this default with a cli command, and we actually already have a cli option we could use here.

Reply:
Do we want to allow the override? When would we want a replace proposal to be sent to a different topic?

Reply:
I can envision some cases, like when some community members request a node replacement in some forum thread, and would like to link to the specific forum post. However, in that case we probably don't want to modify the forum post upon proposal submission - especially if it's one of NP/community member posts.